Tests/FluentAssertions.Specs/Specialized/DelegateAssertionSpecs.cs

```diff
@@ -34,4 +36,36 @@ public void When_injecting_a_null_clock_it_should_throw()
         act.Should().ThrowExactly<ArgumentNullException>()
             .WithParameterName("clock");
     }
+
+    [Fact]
+    public void If_a_function_is_supposed_to_throw_an_exact_type_of_an_exception_while_wrapped_into_an_assertion_scope_it_do_not_proceed_when_the_exception_is_null()
+    {
+        // Arrange
+        var a = () => { };
+
+        // Act
+        using var scope = new AssertionScope();
+        a.Should().ThrowExactly<InvalidOperationException>();
+        var failures = scope.Discard();
+
+        // Assert
+        failures.Should().ContainSingle()
+            .Which.Should().Match("*InvalidOperationException*no exception*");
+    }
+
+    [Fact]
+    public async Task If_an_async_function_is_supposed_to_throw_an_exact_type_of_an_exception_while_wrapped_into_an_assertion_scope_it_do_not_proceed_when_the_exception_is_null()
```

Comment:
🔧 This test belongs in `TaskAssertionSpecs`. This class uses nested classes, which makes it possible to keep the test names shorter